### DISCUSSION TITLE
Dealing with key conflicts

### DISCUSSION THREAD

**Discussion by FeldrinH** [livebud/bud#174] [Q&A]
If I understand the documentation correctly, if your request data has a key conflict (for example the request body and query parameters both contain an 'id' key), then you can only ever get one of the values for the conflicting key.

With that being the case, what should you do if you need to access both values separately (for example you need to have both an 'id' query parameter and an 'id' field in the request body and they have different values and are used for different things) and for whatever reason you cannot rename one of them (external constraints on API design, need to maintain backwards/forwards compatibility, etc)?

**Comment by matthewmueller** [comment]
Thanks for your comment @FeldrinH! That's a great use case I hadn't considered. For now, I'd probably punt on this issue since Bud is primarily targeting greenfield applications. 

That being said, it's reasonable that there's some way to support this in the future. I'd probably look to Rails for the answer since combining the various parts of the request into a single map was inspired by Rails.
